src/Hamiltonians/ExtendedHubbardMom1D.jl

```diff
@@ -0,0 +1,128 @@
+"""
+struct ExtendedHubbardMom1D{TT,M,AD<:AbstractFockAddress,U,V,T,BOUNDARY_CONDITION} <: AbstractHamiltonian{TT}
+    address::AD # default starting address, should have N particles and M modes
+    ks::SVector{M,TT} # values for k
+    kes::SVector{M,TT} # values for kinetic energy
+end
+
+function ExtendedHubbardMom1D(
+    address::SingleComponentFockAddress;
+    u=1.0, v=1.0, t=1.0, dispersion = hubbard_dispersion, boundary_condition = 0.0
+)
+    M = num_modes(address)
+    U, V, T= promote(float(u), float(v), float(t))
+    step = 2π/M
+    if isodd(M)
+        start = -π*(1+1/M) + step
+    else
+        start = -π + step
+    end
+    kr = range(start; step = step, length = M)
+    ks = SVector{M}(kr)
+    kes = SVector{M}(dispersion.(T , kr .+ boundary_condition))
+    return ExtendedHubbardMom1D{typeof(U),M,typeof(address),U,V,T,boundary_condition}(address, ks, kes)
+end
+
+function Base.show(io::IO, h::ExtendedHubbardMom1D)
+    compact_addr = repr(h.address, context=:compact => true) # compact print address
+    print(io, "ExtendedHubbardMom1D($(compact_addr); u=$(h.u), v=$(h.v), t=$(h.t), boundary_condition=$(h.boundary_condition))")
+end
+
+function starting_address(h::ExtendedHubbardMom1D)
+    return h.address
+end
+
+dimension(::ExtendedHubbardMom1D, address) = number_conserving_dimension(address)
+
+LOStructure(::Type{<:ExtendedHubbardMom1D{<:Real}}) = IsHermitian()
+
+Base.getproperty(h::ExtendedHubbardMom1D, s::Symbol) = getproperty(h, Val(s))
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:ks}) = getfield(h, :ks)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:kes}) = getfield(h, :kes)
+Base.getproperty(h::ExtendedHubbardMom1D, ::Val{:address}) = getfield(h, :address)
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,U}, ::Val{:u}) where {U} = U
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,V}, ::Val{:v}) where {V} = V
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,T}, ::Val{:t}) where {T} = T
+Base.getproperty(h::ExtendedHubbardMom1D{<:Any,<:Any,<:Any,<:Any,<:Any,<:Any,BOUNDARY_CONDITION}, 
+    ::Val{:boundary_condition}) where {BOUNDARY_CONDITION} = BOUNDARY_CONDITION
+
+ks(h::ExtendedHubbardMom1D) = getfield(h, :ks)
+
+# standard interface function
+function num_offdiagonals(ham::ExtendedHubbardMom1D, address::SingleComponentFockAddress)
+    singlies, doublies = num_singly_doubly_occupied_sites(address)
+    return num_offdiagonals(ham, address, singlies, doublies)
+end
+
+# 4-argument version
+@inline function num_offdiagonals(ham::ExtendedHubbardMom1D, ::SingleComponentFockAddress, singlies, doublies)
+    M = num_modes(ham)
+    return singlies * (singlies - 1) * (M - 2) + doublies * (M - 1)
+end
+
+@inline function momentum_transfer_diagonal(
+    h::ExtendedHubbardMom1D{<:Any,M,<:BoseFS}, map
+) where {M}
+    return h.u / 2M *momentum_transfer_diagonal(map) 
+    + h.v/ M * dot(cos.(h.ks), map) * extended_momentum_transfer_diagonal(map,M) 
```

Comment:
```suggestion
    + h.v/ M * dot(cos.(h.ks), map) * extended_momentum_transfer_diagonal(map, (2*π)/M) 
```
See comment on `extended_momentum_transfer_diagonal`!